Redesign draft for settings pane and proper UX for Mac

tucnak opened the pull request:
This pull request is capable of:
- **Repairing settings UX on Mac (#1594)**
- **Turning scroll areas in settings panes to vertical scrolling only**
- **Proposing a redesign of settings pane (#1069)**

Here are some neat screenshots:

![screen shot 2015-05-11 at 12 28 50 am](https://cloud.githubusercontent.com/assets/934682/7556397/d867e18e-f776-11e4-90ee-e98d814b38bf.png)
![screen shot 2015-05-11 at 12 39 27 am](https://cloud.githubusercontent.com/assets/934682/7556398/d8690f3c-f776-11e4-8050-b21f107d743e.png)
![screen shot 2015-05-11 at 12 39 46 am](https://cloud.githubusercontent.com/assets/934682/7556399/d883226e-f776-11e4-9d56-c002c904d6dd.png)

Disclaimer: I am still not sure about possible regressions on Linux/Windows, since I haven't got an occasion to test this patch out on these platforms.

tucnak committed:
Redesign draft for settings pane and proper UX for Mac

zetok commented:
![image](https://cloud.githubusercontent.com/assets/3148759/7556437/b3a4a40e-f767-11e4-9664-6c3268cea1a9.png)
↑ with tiny window, I assume that it's intentional that stuff covers other stuff?

---

![:3](https://cloud.githubusercontent.com/assets/3148759/7556445/1e0e934a-f768-11e4-8503-1f6d5fc88709.png)
↑ steps to reproduce:
1. Switch to other settings tab, e.g. `Privacy`
2. minimize qTox window
3. switch to `General` and switch back to `Privacy`
4. change window size back to previous
5. switch to `General` - now it looks as if window was still smaller

Reproducible: Always
Reproducible under i3wm

---

Aside from that, it really makes settings look nicer :3

tux3 commented:
There's also definitely something wrong in the CSS, there shouldn't be a white background behind the QCheckBoxes, on any platform. It's not very noticeable on Linux/Windows, but it's there too.

zetok commented on src/widget/form/settings/verticalonlyscroller.cpp:
```diff
@@ -0,0 +1,31 @@
+/*
+    Copyright (C) 2014 by Project Tox <https://tox.im>
```
> `2014`

?

tucnak committed:
Resolves issues reported in PR before and some BEAST features:

+ Works great on little boxes
+ Fixed the bug spotted by @zetok
+ Smiles are centered under the smiles combo box
+ Namings for layouts

tucnak commented:
The smiley thing from the last commit is so goddamn cool!1
![](https://cloud.githubusercontent.com/assets/934682/7556580/5d04fea6-f77f-11e4-9fec-92bbf57af197.png)

zetok commented:
![emoticons](https://cloud.githubusercontent.com/assets/3148759/7556598/c64cc2a0-f76f-11e4-9b95-b3915b6a9911.png)
↑ is centring emoticons working as intended?

tucnak committed:
Fixing smiley alignment and wrong copyright years

tucnak commented:
Alignment was broken indeed. Hopefully, last commit has fixed it.

tux3 commented:
Okay, it looks good to me.

ProMcTagonist commented:
1) go to Settings > Privacy
2) make the qTox window very small
3) change to General
4) change to a contact or Profile view
5) make qTox big again
6) go to Settings > Privacy
7) it's small

agilob commented:
@ProMcTagonist can't reproduce.

tux3 commented:
@ProMcTagonist I can't repro either

ProMcTagonist commented:
Tested more. When you make it small, make the friends list wider too to squish it more. When I don't do that everything works fine. When I do, stuff stays small.
